arch/x86/core/intel64/locore.S

```diff
@@ -157,8 +157,13 @@ __start:
 	rep stosl
 
 	movl $x86_cpuboot, %ebp		/* BSP is always logical CPU id 0 */
-	movl %ebx, __x86_cpuboot_t_arg_OFFSET(%ebp)	/* multiboot info */
-
+	/* Inserting boot type */
+	movl $MULTIBOOT_BOOT_TYPE, \
+		(__x86_cpuboot_t_arg_OFFSET + \
+		 __x86_boot_arg_t_boot_type_OFFSET)(%ebp)
+	/* and multiboot info */
+	movl %ebx, (__x86_cpuboot_t_arg_OFFSET + \
+		    __x86_boot_arg_t_arg_OFFSET)(%ebp)
 
```

Comment:
To me, this looks to be inserting boot type and multiboot info at the same location. They have the same offsets from %edp, or am I misunderstanding this?

Reply:
oops, I see the difference now. Sorry!